web/src/engine/websites/MadaraDex.ts

```diff
@@ -0,0 +1,55 @@
+import { Tags } from '../Tags';
+import icon from './MadaraDex.webp';
+import { type Chapter, DecoratableMangaScraper, Page, type MangaScraper } from '../providers/MangaPlugin';
+import * as Madara from './decorators/WordPressMadara';
+import { Fetch, FetchCSS, FetchRequest } from '../FetchProvider';
+import type { Priority } from '../taskpool/TaskPool';
+import DeProxify from '../transformers/ImageLinkDeProxifier';
+
+function ChapterPageExtractor(this: MangaScraper, image: HTMLImageElement): string {
+    const url = image.dataset?.src || image.dataset?.lazySrc || image.srcset || image.src;
+    const uri = new URL(url.trim(), this.URI);
+    return uri.href;
+}
+
+@Madara.MangaCSS(/^{origin}\/title\/[^/]+\/$/, 'meta[property="og:title"]:not([content*="Read "])')
+@Madara.MangasMultiPageAJAX()
+@Madara.ChaptersSinglePageAJAXv2()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('madaradex', 'MadaraDex', 'https://madaradex.org', Tags.Media.Manhwa, Tags.Language.English, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const uri = new URL(chapter.Identifier, this.URI);
+        const request = new FetchRequest(uri.href);
+        const data = await FetchCSS<HTMLImageElement>(request, 'div.page-break img');
+        return data.map(element => {
+            const link = new URL(ChapterPageExtractor.call(this, element), request.url);
+            let referer = link.searchParams.get('domain') ?? this.URI.origin;
+            referer += '/';
+            return new Page(this, chapter, DeProxify(link), { Referer: referer });
```

Comment:
Madaradex is a very specific DeProxifier that is not used in any other website.
De-Proxification should be done within this method and not via the common `ImageLinkdeProxifier`.